### PR TITLE
feat: Add field to RawFrame to control instruction address adjustment

### PR DESCRIPTION
**Pull request by loewenheim**
This implements the `symbolicator` part of https://github.com/getsentry/rfcs/pull/43. It adds a new field `instruction_addr_needs_adjustment: Option<bool>` to `RawFrame` which is then used to decide whether the frame's instruction address needs to be adjusted to account for an off-by-one error. Internally, this field is mapped to a new enum `AdjustInstructionAddr` which should make the intended behavior clearer than using a bare `Option<bool>`.

Some open questions:
* The functions `AdjustInstructionAddr::default_for_thread` and `AdjustInstructionAddr::for_frame` are very short and only used in one place each; I created dedicated functions for clarity and documentation purposes. Should we inline these instead?
* What should be the value of `instruction_addr_needs_adjustment` in the frame returned by `symbolicate_native_frame`? Should we use the value from the input frame? Should we just use `None`?

### CODE REVIEW

**Comment by loewenheim** [comment]
Another question: should we make `AdjustInstructionAddr` public for documentation purposes?

File: crates/symbolicator-service/src/services/symbolication/mod.rs [hunk 1]
```diff
@@ -340,6 +397,7 @@ fn symbolicate_native_frame(
                 package: lookup_result.object_info.raw.code_file.clone(),
                 addr_mode: lookup_result.preferred_addr_mode(),
                 instruction_addr,
+                instruction_addr_needs_adjustment: todo!(),
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/symbolication/mod.rs:400]: I think we should just copy over the incoming data here

File: crates/symbolicli/src/main.rs [hunk 2]
```diff
@@ -679,6 +679,7 @@ mod event {
         Some(RawFrame {
             addr_mode: value.addr_mode,
             instruction_addr: value.instruction_addr?,
+            instruction_addr_needs_adjustment: todo!(),
```
Comment (Swatinem) [inline crates/symbolicli/src/main.rs:682]: I would use `None` here

**Review by Swatinem — COMMENTED** [review]
lgtm

**Review by Swatinem — APPROVED** [review]

**Pull request merged**